src/analysis/rule_extraction.jl

```diff
@@ -77,16 +77,97 @@ end
 """
     print_rules(rules::Vector{Rule})
 
-Print all rules in a human readable way. SIRUS also has this functionality but 1) it only 
-prints once, when the model is fitted and 2) this function prints the rules in a more 
+Print all rules in a human readable way. SIRUS also has this functionality but 1) it only
+prints once, when the model is fitted and 2) this function prints the rules in a more
 compact form.
 
 # Arguments
 - `rules` : Vector of Rule objects
 """
-function print_rules(rules::Vector{Rule{Vector{Float64},Vector{Vector{Any}}}})
+function print_rules(rules::Vector{Rule{Vector{Vector},Vector{Float64}}})
     condition(rule) = [c == :L ? "$(c[1]) < $(c[3])" : "$(c[1]) ≥ $(c[3])" for c in rule.condition]
     consequent(rule) = " then $(rule.consequent[1]) else $(rule.consequent[2])\n"
     rule_string(rule) = "if " * join(condition(rule), " ") * consequent(rule)
     print(join([rule_string(rule) for rule in rules]))
 end
+
+
+"""
+    cluster_rules(clusters::Vector{T}, X::DataFrame, outcomes::AbstractMatrix{F}, max_rules::Int64; n_trees::Int64=1000) where {T<:Integer,F<:Real}
+
+Use SIRUS package to extract rules from time series clusters based on some summary metric (default is median)
+
+# Arguments
+- `clusters` - Vector of cluster indexes for each scenario outcome
+- `X::DataFrame` - DataFrame with factors to be used as input by SIRUS
+- `outcomes` - Matrix of some metric over time for all scenarios
+- `max_rules` - Maximum number of rules, to be used as input by SIRUS
+- `n_trees` - Number of trees to be created by SIRUS algorithm
+
+# Returns
+A StableRules object (implemented by SIRUS).
+
+# References
+1. Huijzer, R.
+   rikhuijzer/SIRUS.jl [Computer software].
+   https://doi.org/10.5281/zenodo.7875577
+
+2. Bénard, C., Biau, G., Da Veiga, S., & Scornet, E. 2021.
+   Sirus: Stable and interpretable rule set for classification.
+   Electron. J. Statist. 15 (1) 427 - 505.
+   https://doi.org//10.1214/20-EJS1792
+"""
+function cluster_rules(clusters::Vector{T}, X::DataFrame, outcomes::AbstractMatrix{F},
+    max_rules::T; n_trees::T=1000) where {T<:Int64,F<:Real}
+    # Find robust cluster index
+    rc = robust_cluster(clusters, outcomes)
+    y = clusters .== rc
+
+    # Set seed and Random Number Generator
+    seed = 123
```

Comment:
I think best to make this a keyword argument with default `123`.

One approach to convergence analysis is to change the seed value to see how big of a difference in conclusions you get so it would be useful to be able to change the seed

Reply:
I agree, that's better.